src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -98,7 +98,11 @@ internal CompletionSource(ITextView textView, IThreadingContext threadingContext
 
             if (!document.Project.Solution.Workspace.Options.GetOption(CompletionOptions.BlockForCompletionItems, service.Language))
             {
-                _textView.Options.GlobalOptions.SetOptionValue(NonBlockingCompletionEditorOption, true);
+                // The Editor supports the option per textView.
+                // There could be mixed desired behavior per textView and even per same completion session.
+                // The right fix would be to send this information as a result of the method. 
+                // Then, the Editor would choose the right behavior for mixed cases.
```

Comment:
Should this link to a tracking bug if there is one? If there isn't one then one more comment of "but we don't think that's really a scenario so this is fine" might be useful.